extensions/replication/ReplicationAPI.js

```diff
@@ -66,24 +68,46 @@ class ReplicationAPI {
      * @return {undefined}
      */
     static sendDataMoverAction(producer, action, log, cb) {
-        const { bucket, key } = action.getAttribute('target');
-        const kafkaEntries = [{ key: `${bucket}/${key}`,
-                                message: action.toKafkaMessage() }];
-        producer.sendToTopic(dataMoverTopic, kafkaEntries, (err, reports) => {
+        const { accountId, bucket, key, version } = action.getAttribute('target');
+        const kafkaEntries = [{
+            key: `${bucket}/${key}`,
+            message: action.toKafkaMessage(),
+        }];
+        let topic = dataMoverTopic;
+        const toLocation = action.getAttribute('toLocation');
+        const locationConfig = locations[toLocation];
+        if (!locationConfig) {
+            const errorMsg = 'could not get destination location configuration';
+            log.error(errorMsg, { method: 'ReplicationAPI.sendDataMoverAction' });
+            return cb(new Error(errorMsg));
+        }
+        if (locationConfig.isCold) {
+            topic = `${coldStorageArchiveTopicPrefix}${toLocation}`;
+            const { reqId } = action.getContext();
+            const message = {
+                accountId,
+                bucketName: bucket,
+                objectKey: key,
+                objectVersion: version,
+                requestId: reqId,
+            };
+            kafkaEntries[0].message = JSON.stringify(message);
```

Comment:
Instead of reaching inside the array and assuming its size, just create `kafkaEntry` instead and send `[kafkaEntry]`.